src/core/ext/filters/client_channel/resolver/dns/event_engine/event_engine_client_channel_resolver.cc

```diff
@@ -0,0 +1,517 @@
+               .value_or(true)),
+      enable_srv_queries_(channel_args()
+                              .GetBool(GRPC_ARG_DNS_ENABLE_SRV_QUERIES)
+                              .value_or(false)),
+      // TODO(yijiem): decide if the ares channel arg timeout should be reused.
+      query_timeout_ms_(std::chrono::milliseconds(
+          std::max(0, channel_args()
+                          .GetInt(GRPC_ARG_DNS_ARES_QUERY_TIMEOUT_MS)
+                          .value_or(GRPC_DNS_DEFAULT_QUERY_TIMEOUT_MS)))),
+      event_engine_(channel_args().GetObjectRef<EventEngine>()) {}
+
+OrphanablePtr<Orphanable> EventEngineClientChannelDNSResolver::StartRequest() {
+  return MakeOrphanable<EventEngineDNSRequestWrapper>(
+      Ref(DEBUG_LOCATION, "dns-resolving"),
+      event_engine_->GetDNSResolver({/*dns_server=*/authority()}));
+}
+
+// ----------------------------------------------------------------------------
+// EventEngineDNSRequestWrapper definition
+// ----------------------------------------------------------------------------
+
+EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
+    EventEngineDNSRequestWrapper(
+        RefCountedPtr<EventEngineClientChannelDNSResolver> resolver,
+        std::unique_ptr<EventEngine::DNSResolver> event_engine_resolver)
+    : resolver_(std::move(resolver)),
+      event_engine_resolver_(std::move(event_engine_resolver)) {
+  // Locking to prevent completion before all records are queried
+  MutexLock lock(&on_resolved_mu_);
+  hostname_handle_ = event_engine_resolver_->LookupHostname(
+      [self = Ref(DEBUG_LOCATION, "OnHostnameResolved")](
+          absl::StatusOr<std::vector<EventEngine::ResolvedAddress>> addresses) {
+        self->OnHostnameResolved(std::move(addresses));
+      },
+      resolver_->name_to_resolve(), kDefaultSecurePort,
+      resolver_->query_timeout_ms_);
+  GRPC_EVENT_ENGINE_RESOLVER_TRACE(
+      "DNSResolver::%p Started resolving hostname. Handle::%s", resolver_.get(),
+      HandleToString(*hostname_handle_).c_str());
+  if (resolver_->enable_srv_queries_) {
+    srv_handle_ = event_engine_resolver_->LookupSRV(
+        [self = Ref(DEBUG_LOCATION, "OnSRVResolved")](
+            absl::StatusOr<std::vector<EventEngine::DNSResolver::SRVRecord>>
+                srv_records) { self->OnSRVResolved(std::move(srv_records)); },
+        resolver_->name_to_resolve(), resolver_->query_timeout_ms_);
+    GRPC_EVENT_ENGINE_RESOLVER_TRACE(
+        "DNSResolver::%p Started resolving SRV. Handle::%s", resolver_.get(),
+        HandleToString(*srv_handle_).c_str());
+  }
+  if (resolver_->request_service_config_) {
+    txt_handle_ = event_engine_resolver_->LookupTXT(
+        [self = Ref(DEBUG_LOCATION, "OnTXTResolved")](
+            absl::StatusOr<std::string> service_config) {
+          self->OnTXTResolved(std::move(service_config));
+        },
+        resolver_->name_to_resolve(), resolver_->query_timeout_ms_);
```

Comment:
I just noticed an issue here in terms of the EE API semantics for TXT record lookups.  [gRFC A2](https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md#encoding-in-dns-txt-records) says that we encode the service config in a TXT record whose name is the same as the hostname but that has a `_grpc_config.` prefix.  We're not passing in that prefix here, which implies that we're assuming that the EE impl will add this prefix.  However, I think we should actually add the prefix here instead of in the EE impl, because the EE TXT lookup API is intended to be a generic mechanism for DNS TXT lookups, not something specific to gRPC service config.

CC @yijiem

Reply:
Totally agree, I missed that. I've added that to the client channel resolver.